src/components/05_pages/NodeForm/index.js

```diff
@@ -41,44 +42,18 @@ class NodeForm extends React.Component {
       }),
       PropTypes.bool,
     ]),
+    onChange: PropTypes.func,
   };
 
   static defaultProps = {
     schema: false,
     uiSchema: false,
+    onChange: () => {},
   };
 
-  state = {};
-
-  static getDerivedStateFromProps(props, prevState) {
```

Comment:
I moved the state calculation to ```componentDidMount/componentDidUpdate``` due to https://reactjs.org/docs/react-component.html#componentdidupdate especially because ```getDerivedStateFromProps``` causes an infinite loop